### DISCUSSION TITLE
The choice of C not Rust for the transpile target

### DISCUSSION THREAD

**Discussion by ken-okabe** [HigherOrderCO/HVM1#27] [Q&A] [answered]
Just from curiosity, I wonder what's the reason would be to have chosen C not Rust as the transpile/compile target language because the HVM itself is written in Rust.

Is Rust redundantly too strict for the memory leak safety and rather hazard to compose the efficient code after the theoretical guarantee of the memory-safe code from GC-less of Symmetric Interaction Calculus?

**Comment by zicklag** [comment]
There's an issue that has some discussion on the topic that should answer your questions: https://github.com/Kindelia/HVM/issues/69.

**Reply by steinerkelvin** [reply]
Btw, I intend in the future to replace as much redundant code as possible on the generated C by binding it with the Rust code, and generating just the relevant part that is "dynamic".

**Comment by VictorTaelin** [comment, marked as answer]
It compiles to Rust now!

**Reply by zicklag** [reply]
Awesome! I've entertained the idea of using HVM for scripting, maybe plugging into Bevy for Rendering.

With HVM compiling to Rust, that would give us the possibility to load non-built-in scripts as runtime-compiled HVM, and for built-in scripts, we could compile to Rust, and embed it!

**Reply by steinerkelvin** [reply]
Oh, so you intend to user it with a game engine? This is hype!

**Reply by zicklag** [reply]
I'm thinking about it. :)

I'm not sure if I'll be able to or not, but it's something I'm interested in trying.
